src/circuit/gadget/ecc/chip.rs

```diff
@@ -0,0 +1,280 @@
+use super::EccInstructions;
+use crate::circuit::gadget::utilities::{copy, CellValue, Var};
+use crate::constants::{self, OrchardFixedBasesFull, ValueCommitV};
+use arrayvec::ArrayVec;
+use ff::{Field, PrimeFieldBits};
+use halo2::{
+    arithmetic::CurveAffine,
+    circuit::{Chip, Layouter},
+    plonk::{Advice, Column, ConstraintSystem, Error, Fixed, Permutation, Selector},
+};
+use std::marker::PhantomData;
+
+// pub(super) mod add;
+// pub(super) mod add_incomplete;
+// pub(super) mod mul;
+// pub(super) mod mul_fixed;
+// pub(super) mod witness_point;
+// pub(super) mod witness_scalar_fixed;
+
+/// A curve point represented in affine (x, y) coordinates. Each coordinate is
+/// assigned to a cell.
+#[derive(Clone, Debug)]
+pub struct EccPoint<C: CurveAffine> {
+    /// x-coordinate
+    pub x: CellValue<C::Base>,
+    /// y-coordinate
+    pub y: CellValue<C::Base>,
```

Comment:
I agree, but IMHO there should be accessors. We are not trying to hide that this is an affine short-Weierstrass point representation with (0, 0) as the zero point. (The circuit depends heavily on that.) We're only trying to avoid construction of invalid point representations.

Reply:
Should these getters be gadget-level?